string str_data1
string str_data2
int16 int_data